newsfragments/141.bugfix

```diff
@@ -0,0 +1 @@
+Upload LocalSnapshot instances to Tahoe (yielding RemoteSnapshot instances)
```

Comment:
It took me a while to understand the scope of this PR.  It's *tooling* for uploading snapshots but on its own it won't actually cause magic-folder to upload anything.  So maybe it's a "minor" fragment (it's 800+ lines so it doesn't feel all that minor but I have trouble thinking about how to frame this as a meaningful user-facing change rather than as internal plumbing that will support a new user-facing feature in the future).

Reply:
Okay.

I don't really care what we label this, but are release-notes always **only** for users? A big percentage of the changes we make aren't going to have use-visible consequences (after all, this is a completely new release anyway). Surely interested developers might read the release notes (and may be curious to know that the internal structure has changed)?

Is the action here to file some other ticket that's "part of" 141 and re-label this to that?

Reply:
> I don't really care what we label this, but are release-notes always only for users?

I think given the likely audience that's what makes the most sense for magic-folders.  If we wanted to share more technical stuff, my guess is we'd want to present it separately from other content.  Maybe we should have a ".tech" fragment type?

I think we do need another ticket - something that captures the technical component of "a snapshot gets uploaded when a file is added".  I don't really have a preference between this PR resolving #141 or that other new ticket - I just want to be sure the remaining work is being tracked as an issue.

Reply:
I filed a few tickets; this one is now for #191 (and adjusted the news entry)